agent/hcp/client/client.go

```diff
@@ -78,6 +99,33 @@ func httpClient(c config.CloudConfig) (*httptransport.Runtime, error) {
 	})
 }
 
+// FetchTelemetryConfig obtains telemetry configuration from the Telemetry Gateway.
+func (c *hcpClient) FetchTelemetryConfig(ctx context.Context) (*TelemetryConfig, error) {
+	clusterID, err := resource.FromString(c.cfg.ResourceID)
+	if err != nil {
+		return nil, err
+	}
+	params := hcptelemetry.NewAgentTelemetryConfigParamsWithContext(ctx).
+		WithLocationOrganizationID(c.resource.Organization).
+		WithLocationProjectID(c.resource.Project).
+		WithClusterID(clusterID.String())
```

Comment:
```suggestion
	clusterResource, err := resource.FromString(c.cfg.ResourceID)
	if err != nil {
		return nil, err
	}
	params := hcptelemetry.NewAgentTelemetryConfigParamsWithContext(ctx).
		WithLocationOrganizationID(c.resource.Organization).
		WithLocationProjectID(c.resource.Project).
		WithClusterID(clusterResource.ID)
```

`FromString` returns a struct with everything: org, project, type, ID. If we `.String()` it we're building up that resource link again `/organization/$organization/project/$project/$type/$id`. We already have org, project covered with the other two params above. We just need `ID` here (the user settable slug that's unique within a project. Eg "consul-cluster" or "fake-cluster" etc)

Reply:
Ahh ok I was misunderstanding! I thought the clusterID was misnamed, and required the full resourceID. Thank you

Reply:
👍 

One small thing, realized we already have resource in `c.resource` on the receiver, no need to parse the string again